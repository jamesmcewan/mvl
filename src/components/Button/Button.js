import React from 'react';
import camelCase from 'lodash.camelcase';
import styled, { css } from 'styled-components';

const StyledButton = styled.button`
  font-size: 16px;
  padding: 13px 6px;
  flex-grow: 1;
  background: transparent;
  color: #6272a4;
  box-sizing: border-box;
  border: 0;

  @media screen and (min-width: 700px) {
    padding: 13px;
    font-size: 26px;
  }

  ${props =>
    props.current &&
    css`
      color: #ff79c6;
    `};
`;

const Button = ({ dateDescriptor, handleClick, current }) => (
  <StyledButton
    {...{current}}
    onClick={e => handleClick(e, camelCase(dateDescriptor))}
  >
    {dateDescriptor}
  </StyledButton>
);

export default Button;
