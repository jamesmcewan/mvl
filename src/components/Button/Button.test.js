import React from 'react';
import { mount } from 'enzyme';
import Button from './Button';
import 'jest-styled-components';

describe('Button', () => {
  it('should render without crashing', () => {
    const wrapper = mount(<Button />);
    expect(wrapper).toHaveLength(1);
  });

  it('should render the correct JSX', () => {
    const wrapper = mount(<Button />);
    expect(wrapper).toMatchSnapshot();
  });
});
