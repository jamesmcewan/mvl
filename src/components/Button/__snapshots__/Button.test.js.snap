// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render the correct JSX 1`] = `
.c0 {
  font-size: 16px;
  padding: 13px 6px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  background: transparent;
  color: #6272a4;
  box-sizing: border-box;
  border: 0;
}

@media screen and (min-width:700px) {
  .c0 {
    padding: 13px;
    font-size: 26px;
  }
}

<Button>
  <styled.button
    onClick={[Function]}
  >
    <button
      className="c0"
      onClick={[Function]}
    />
  </styled.button>
</Button>
`;
